source/flattener.process.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * flatten: Handles edge cases in a clip list to prepare it for rendering. Flattens multiple tracks into a single track and replaces spaces with seconds of blackness.
+ *
+ * INPUT
+ * - ticSortedClips (JSON)
+ * A JSON array of clip metadata. Should be sorted by tic in ascending order. 
+ *
+ * OUTPUT
+ * - A promise that will resolve when the flattened array of clips has been completed, and will return the flattened array.
+ * 
+ */
+module.exports.flatten = function(ticSortedClips)
+{
+    var previousEndTick = 0;
+    for (var i = 0; i < ticSortedClips.length; i++)
+    {
+        var clip = ticSortedClips[i];
+        
+        // For now, naively put all clips on track 0
+        clip.track = 0;
+
+        if (clip.tick > previousEndTick)
```

Comment:
Let's comment some of the logic as we go. :]

    //If this beginning of this clip overlaps with the end of the last clip...